### PR TITLE
Render YARP templates in the build process

### PR DESCRIPTION
**Pull request by k0kubun**

### CODE REVIEW

File: tool/sync_default_gems.rb [hunk 1]
```diff
@@ -600,7 +627,7 @@ def sync_default_gems_with_commits(gem, ranges, edit: nil)
         puts "Remove files added to toplevel: #{toplevels.join(', ')}"
         system(*%w"git rm -r --", *toplevels)
       end
-      tools = changed.select {|f|f.start_with?("test/fixtures/", "test/lib/", "tool/")}
+      tools = changed.select {|f|f.start_with?("test/lib/", "tool/")}
```
Comment (nobu) [inline tool/sync_default_gems.rb]: Ditto.

File: tool/sync_default_gems.rb [hunk 2]
```diff
@@ -371,7 +371,7 @@ def sync_default_gems(gem)
       `git checkout ext/digest/depend ext/digest/*/depend`
     when "set"
       sync_lib gem, upstream
-      cp_r(Dir.glob("#{upstream}/test/*"), "test/set")
+      cp_r("#{upstream}/test", ".")
```
Comment (nobu) [inline tool/sync_default_gems.rb:374]: This revert previous changes,

**Review by nobu — DISMISSED** [review]

File: tool/sync_default_gems.rb [hunk 3]
```diff
@@ -420,7 +419,10 @@ def sync_default_gems(gem)
       cp_r("#{upstream}/include/yarp/.", "yarp")
       cp_r("#{upstream}/include/yarp.h", "yarp")
 
-      rm("yarp/config.h")
+      cp_r("#{upstream}/config.yml", "tool/")
+      cp_r("#{upstream}/templates", "tool/")
```
Comment (nobu) [inline tool/sync_default_gems.rb:423]: Seems copying YARP specific files to the common directory?
Really intentional?

File: tool/sync_default_gems.rb (hunk 2 above)
Comment (k0kubun) [reply tool/sync_default_gems.rb]: thanks :pray: https://github.com/ruby/ruby/pull/8228/commits/c7f3913ca772dfe990f45337cb87edcfb83b3b14

File: tool/sync_default_gems.rb (hunk 1 above)
Comment (k0kubun) [reply tool/sync_default_gems.rb]: I fixed the other one, but it seems like you commented on an outdated diff for this one.

File: tool/sync_default_gems.rb (hunk 3 above)
Comment (k0kubun) [reply tool/sync_default_gems.rb:423]: No, this should live under `tool/yarp`. I'll mark this as a draft until it's fixed.

Comment (k0kubun) [reply tool/sync_default_gems.rb]: I changed my mind a little and put that under `yarp` instead.

**Pull request merged**